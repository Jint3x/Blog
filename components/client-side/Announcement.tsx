import React from "react"


interface Announcement_I {
    title: String,
    desc: String,
    bgColor: String,
    textColor: String,
    headerColor: String,
    borderColor: String,
}

function Announcement(props: Announcement_I) {
    return (
        <div className={`${props.borderColor} ${props.bgColor} p-2 border-solid border-l-4`}>
            <h3 className={`${props.headerColor} text-lg md:text-xl`}>
                {props.title}
            </h3>
            <p className={`${props.textColor} text-base pt-2 pb-2`}>
                {props.desc}
            </p>
        </div>
    )
}


export default Announcement