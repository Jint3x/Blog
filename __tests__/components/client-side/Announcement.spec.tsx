/**
 * @jest-environment jsdom
 */
import Announcement from "../../../components/client-side/Announcement";
import ReactRenderer from "react-test-renderer"
import { screen, render } from "@testing-library/react"


it("Renders an announcement with a header and text", () => {
    render(
        <Announcement
         title={"A special announcement"}
         desc={"This component will contain a special message"}
         bgColor={"text-mainBlue-500"}
         textColor={"text-black"}
         headerColor={"text-white"}
         borderColor={"random-class"}
        />
    );

    let header = screen.getByText("A special announcement");
    let message = screen.getByText("This component will contain a special message");
    
    expect(header.nodeName).toBe("H3");
    expect(message.nodeName).toBe("P");
})


it("Appends the correct classes to the elements", () => {
    render(
        <Announcement
         title={"A special announcement"}
         desc={"This component will contain a special message"}
         bgColor={"bg-black"}
         textColor={"text-black"}
         headerColor={"text-white"}
         borderColor={"random-class"}
        />
    );

    let header = screen.getByText("A special announcement");
    let message = screen.getByText("This component will contain a special message");

    expect(header?.classList).toContain("text-white");
    expect(message?.classList).toContain("text-black");
    expect(header?.parentElement?.classList).toContain("random-class")
    expect(header?.parentElement?.classList).toContain("bg-black")
})


it("Takes a snapshot of the component", () => {
    let tree = ReactRenderer.create(
        <Announcement
         title={"A special announcement"}
         desc={"This component will contain a special message"}
         bgColor={"bg-black"}
         textColor={"text-black"}
         headerColor={"text-white"}
         borderColor={"random-class"}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
})