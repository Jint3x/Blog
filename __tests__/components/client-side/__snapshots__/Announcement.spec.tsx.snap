// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes a snapshot of the component 1`] = `
<div
  className="random-class bg-black p-2 border-solid border-l-4"
>
  <h3
    className="text-white text-lg md:text-xl"
  >
    A special announcement
  </h3>
  <p
    className="text-black text-base pt-2 pb-2"
  >
    This component will contain a special message
  </p>
</div>
`;
